crates/kgraph_utils/src/mca.rs

```diff
@@ -257,17 +262,229 @@ fn compile_payment_method_enabled(
+    let mut pmt_enabled: Vec<dir::DirValue> = Vec::new();
+    if let Some(pmt) = config
+        .connector_configs
+        .get(connector)
+        .or(config.default_configs.as_ref())
+        .map(|inner| inner.0.clone())
+    {
+        for key in pmt.keys().cloned() {
+            match key {
+                PaymentMethodFilterKey::PaymentMethodType(pm) => {
+                    let dir_val_pm = pm.into_dir_value().map(Into::into)?;
+                    pmt_enabled.push(dir_val_pm);
+
+                    let curr = pmt
+                        .get(&PaymentMethodFilterKey::PaymentMethodType(pm))
+                        .map(|country| {
+                            let pm_node = if pm == api_enums::PaymentMethodType::Credit
+                                || pm == api_enums::PaymentMethodType::Debit
+                            {
+                                builder.make_value_node(
+                                    cgraph::NodeValue::Value(dir::DirValue::PaymentMethod(
+                                        dir::enums::PaymentMethod::Card,
+                                    )),
+                                    Some("PaymentMethod"),
+                                    None::<()>,
+                                )
+                            } else {
+                                builder.make_value_node(
+                                    pm.into_dir_value().map(Into::into)?,
+                                    Some("PaymentMethodType"),
+                                    None::<()>,
+                                )
+                            };
+                            compile_graph_for_countries_and_currencies(builder, country, pm_node)
+                        })
+                        .transpose()?;
+
+                    if let Some(country_currency) = curr {
+                        agg_node_id.push((
+                            country_currency,
+                            cgraph::Relation::Positive,
+                            cgraph::Strength::Normal,
+                        ));
+                    }
+                }
+                PaymentMethodFilterKey::CardNetwork(cn) => {
+                    let dir_val_cn = cn.clone().into_dir_value()?;
+                    pmt_enabled.push(dir_val_cn);
+                    let cn_id = builder.make_value_node(
+                        cn.clone().into_dir_value().map(Into::into)?,
+                        Some("CardNetwork"),
+                        None::<()>,
+                    );
+                    let curr = pmt
+                        .get(&PaymentMethodFilterKey::CardNetwork(cn.clone()))
+                        .map(|country| {
+                            compile_graph_for_countries_and_currencies(builder, country, cn_id)
+                        })
+                        .transpose()?;
+
+                    if let Some(_country_currency) = curr {
```

Comment:
I'm kinda surprised clippy didn't suggest this.
```suggestion
                    if curr.is_some() {
```